### PR TITLE
Animation system/ctp 36 animation swapping(Without Asset)

### PR DESCRIPTION
**Pull request by Chenziyes**
Update More advanced animation loading method(by pose)
Implemented switching animation with character state 
Multiple animation 
Guard state
tutorialGame to levelManager

### CODE REVIEW

File: CSC8503/TutorialGame.cpp [hunk 1]
```diff
@@ -549,33 +532,33 @@ GameObject* TutorialGame::AddBonusToWorld(const Vector3& position, const std::st
 	return apple;
 }
 
-GameObject* TutorialGame::AddAnimationTest(const Vector3& position, const std::string& objectName)
-{
-	float meshSize = 1.0f;
-	float inverseMass = 0.5f;
-
-	GameObject* animTest = new GameObject();
```
Comment (benblack90) [inline CSC8503/TutorialGame.cpp:557]: We won't be using tutorial game. Will your code still work?

File: CSC8503CoreClasses/AnimationSystem.cpp [hunk 2]
```diff
@@ -24,87 +27,160 @@ void AnimationSystem::Clear()
...
+				o->GetRenderObject()->SetFrameMatricesVec(frameMatricesVec);
+				
+				frameMatricesVec.clear();
+				
+				
+				
+
+				if (o->GetName() == "Guard") {
+					mGuardList.emplace_back((GuardObject*)o);
+					/*std::cout << "find the guard" << std::endl;*/
```
Comment (benblack90) [inline CSC8503CoreClasses/AnimationSystem.cpp:97]: Best to delete commented out code. If you still need it for some reason, mark it with your name, so we know who to chase if it never gets done ;)

**Review by benblack90 — APPROVED** [review]
Make sure you clean up some of the comments, and then import the assets quickly after this, so the build does not break

File: CSC8503/TutorialGame.cpp (hunk 1 above)
Comment (Chenziyes) [reply CSC8503/TutorialGame.cpp:557]: Nope, I will delete it in next time.

File: CSC8503CoreClasses/AnimationSystem.cpp (hunk 2 above)
Comment (Chenziyes) [reply CSC8503CoreClasses/AnimationSystem.cpp:97]: Got it!

File: CSC8503CoreClasses/AnimationSystem.cpp [hunk 3]
```diff
@@ -24,87 +27,160 @@ void AnimationSystem::Clear()
...
-void AnimationSystem::Update(float dt)
+void AnimationSystem::Update(float dt, vector<GameObject*> UpdatableObjects,std::map<std::string,MeshAnimation*> preAnimationList)
 {
-	
-	UpdateAllAnimationObjects(dt);
-	UpdateMaterials();
+	UpdateCurrentFrames(dt);
+	UpdateAnimations(preAnimationList);
+	UpdateAllAnimationObjects(dt, UpdatableObjects);
+	/*UpdateMaterials();*/
```
Comment (erendgrmnc) [inline CSC8503CoreClasses/AnimationSystem.cpp:35]: minor: if we want to merge comment line best to add a todo!

**Review by erendgrmnc — APPROVED** [review]

**Pull request merged**